client-src/rundump.c

```diff
@@ -172,15 +175,35 @@ main(
 # endif
 #endif
 
-
     /*
      * Build the array
-     */
-
-    g_ptr_array_add(array, g_strdup(dump_program));
-
-    for (i = 1; argv[i]; i++)
-        g_ptr_array_add(array, quote_string(argv[i]));
+     */	 
+	g_ptr_array_add(array, g_strdup(dump_program));
+	
+	// No need to deallocate "prgName", as strrchr is not returning newly allocated memory
+	// but just returning an offset to the string.
+	prgName = g_strrstr(dump_program, "/");
+	if (prgName == NULL)
+	{
+		prgName = dump_program;
+	}
+	else
+	{
+		prgName++;
+	}
+	if (g_strcmp0(prgName, "dump") == 0) {
+		if (validate_dump_option(argc, argv) != 0) {
+			error(" ");
+		}
+	}
+	else if (g_strcmp0(prgName, "xfsdump") == 0) {
+		if (validate_xfsdump_options(argc, argv) != 0) {
+			error(" ");
+		}
+	}
```

Comment:
It would probably be better to use the preprocessor conditionals (`#if defined(DUMP)`, etc.) here instead of trying to figure out which program is in use at runtime.  If an Amanda install is set up with, say, XFSDUMP=/opt/fs/old-xfsdump, then the code above would never call `validate_xfsdump_options`, even though it's clear that it's driving xfsdump.

Reply:
Hello Dustin,
Thank you for your review comment.
Yes, I will make the changes accordingly.